src/pykeen/evaluation/rank_based_evaluator.py

```diff
@@ -182,12 +182,18 @@ class RankBasedEvaluator(Evaluator):
 
     def __init__(
         self,
-        ks: Optional[Iterable[int]] = None,
+        ks: Optional[Iterable[Union[int, float]]] = None,
         filtered: bool = True,
     ):
         super().__init__(filtered=filtered)
         self.ks = tuple(ks) if ks is not None else (1, 3, 5, 10)
+        for k in self.ks:
+            if isinstance(k, float) and not (0 < k < 1):
```

Comment:
@mberr Maybe it would be good to make sure that if one of the `ks` is a float that all of them need to be `floats` or do we want allow mixed cases?